### PR TITLE
Adding progress callback to provide feedback by writing dots . to the…

### PR DESCRIPTION
**Pull request by BobBrownConsulting**
… command line (#145))

### CODE REVIEW

File: src/main/python/ttconv/tt.py [hunk 1]
```diff
@@ -39,6 +39,10 @@
 
 LOGGER = logging.getLogger(__name__)
```
Comment (BobBrownConsulting) [inline src/main/python/ttconv/tt.py:41]: Writing out the "." should probably be throttled by time or something else such that we don't fill up our terminal

Comment (palemieux) [reply src/main/python/ttconv/tt.py:41]: What about the callback returning a value between 0 and 1?

**Review by palemieux — CHANGES_REQUESTED** [review]

File: src/main/python/ttconv/tt.py (hunk 1 above)
Comment (BobBrownConsulting) [reply src/main/python/ttconv/tt.py:41]: Done

File: src/main/python/ttconv/imsc/elements.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def is_instance(xml_elem) -> bool:
     raise NotImplementedError
 
   @staticmethod
-  def from_xml(parent_ctx, xml_elem):
+  def from_xml(parent_ctx, xml_elem, progress_callback = None):
```
Comment (palemieux) [inline src/main/python/ttconv/imsc/elements.py:99]: Any reason for not moving `progress_callback` into `parent_ctx`?

**Review by valnoel — COMMENTED** [review]
@BobBrownConsulting @palemieux 
I'd suggest that the default `progress_callback` parameter value to be an empty function, so that we don't have to check whether the callback is set or None.
Something like:
```python
def to_model(..., progress_callback=lambda _: None):
    ...
    progress_callback(0.5)
    ...
```
instead of:
```python
def to_model(..., progress_callback=None):
    ...
    if progress_callback:
        progress_callback(0.5)
    ...
```

File: src/main/python/ttconv/imsc/elements.py (hunk 2 above)
Comment (BobBrownConsulting) [reply src/main/python/ttconv/imsc/elements.py:99]: I don't think putting this in the parent_ctx is the right thing to do. There isn't the same parallel for the writer (from_model) 

Also, the parent context is currently owned/created in class TTElement(TTMLElement): from_xml. 

The caller who pass in the progress bar is the tt.py layer which passes it into reader to_model which passes in a None for parent context via the call imsc_elements.TTElement.from_xml

Give that, making the parent context own it, seems odd to me.

**Comment by BobBrownConsulting** [comment]
@palemieux I've updated to master and resolve conflicts. Please do a final pass for the review.

**Review by palemieux — APPROVED** [review]

**Pull request merged**